pkg/util/file_indexer.go

```diff
@@ -83,6 +86,77 @@ func resolveFilePath(directory string) (string, error) {
 	return directory, nil
 }
 
+// gitignoreFilePath gives the filepath of the .gitignore file in the contex
+func gitignoreFilePath(directory string) (string, error) {
+	directoryFi, err := os.Stat(filepath.Join(directory))
+	if err != nil {
+		return "", err
+	}
+
+	switch mode := directoryFi.Mode(); {
+	case mode.IsDir():
+		// do directory stuff
+		return filepath.Join(directory, ".gitignore"), nil
+	case mode.IsRegular():
+		// do file stuff
+		// for binary files
+		return filepath.Join(filepath.Dir(directory), ".gitignore"), nil
+	}
+
+	return directory, nil
+}
+
+// Check .gitignore file exists or not
+// Add odo-file-index.json entry to .gitignore
+func checkGitignoreFile(directory string) {
+
+	ignoreFile, err := gitignoreFilePath(directory)
+	if err != nil {
+		log.Fatalf("file does not exist: %s", err)
+	}
+
+	_, err = os.Stat(ignoreFile)
+
+	if err == nil {
+		file, err := os.OpenFile(ignoreFile, os.O_APPEND|os.O_RDWR, 0600)
```

Comment:
have you tested this on windows?

Reply:
No not yet. Will test and update the result here.